### PR TITLE
Use fastify-warning for handling deprecation warnings

### PR DESCRIPTION
**Pull request by delvedor**
Continuation of https://github.com/fastify/fastify/pull/2339.
Please, take also a look at https://github.com/fastify/fastify-warning/pull/1.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: fastify.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ const {
   kPluginNameChain
 } = require('./lib/symbols.js')
 
+require('./lib/warnings')
```
Comment (mcollina) [inline fastify.js:29]: why is this needed?

Comment (delvedor) [reply fastify.js:29]: For load the warnings in memory. They are still declared in that file :)

Comment (jsumners) [reply fastify.js:29]: Is this how our current implementation works? The side effect nature of this is not great.

Comment (mcollina) [reply fastify.js:29]: I don't think this `require` is needed at all.

Comment (delvedor) [reply fastify.js:29]: @mcollina I fear I am missing something then. How would you load the warnings then?

@jsumners yes, currently the module is a singleton, I didn’t apply any changes from the original Fastify code. I was thinking about exposing a builder function tho.

Comment (mcollina) [reply fastify.js:29]: I think you can remove this line and all tests will pass. The warnings will be loaded by the other files that needs `emitWarning`.

Comment (delvedor) [reply fastify.js:29]: Nope, they will not :)
`fastify-warning` at the moment is a singleton, and if we remove this require, then the `createWaring` api will never be called. As result, `emitWarning` will start failing because the deprecations codes we are using have not been registered.

**Comment by delvedor** [comment]
@mcollina @jsumners PTAL. Also review https://github.com/fastify/fastify-warning/pull/1.

File: docs/Plugins-Guide.md [hunk 2]
```diff
@@ -330,6 +331,16 @@ const CustomError = createError('ERROR_CODE', 'message')
 console.log(new CustomError())
 ```
 
+<a name="emit-warnings"></a>
+## Emit Warnings
+If you want to deprecate an API, or you want to warn the user about a specific use case, you can use the [`fastify-warning`](https://github.com/fastify/fastify-warning) module.
+
+```js
+const warning = require('fastify-warning')()
```
Comment (mcollina) [inline docs/Plugins-Guide.md:339]: It would be good to explain why this is a factory.

**Comment by mcollina** [comment]
@delvedor what's the state of this?

**Comment by delvedor** [comment]
@mcollina Sorry, got caught with other stuff :)
I'll try to finish this this weekend.

**Comment by delvedor** [comment]
> It would be good to explain why this is a factory.

It is already explained in the module documentation :)
